### PR TITLE
Add inverse mappings for exif tags

### PR DESCRIPTION
**Pull request by melonmouse**
A nicer solution would be to switch to an enum with the right values, but this is the quick win.

Tested locally, ran through pycodestyle. To me it seems the code is trivial enough to not warrant adding a unit test.

### CODE REVIEW

File: docs/reference/ExifTags.rst [hunk 1]
```diff
@@ -4,25 +4,45 @@
...
 
         >>> from PIL.ExifTags import GPSTAGS
         >>> GPSTAGS[20]
         'GPSDestLatitude'
+        
+.. py:data:: GPS_CODES
+    :type: dict
+
+    The GPS_CODES dictionary maps descriptive string names to 8-bit integer EXIF
+    gps enumerations. For instance:
```
Comment (radarhere) [inline docs/reference/ExifTags.rst:44]:
```suggestion
    GPS enumerations. For instance:
```

**Comment by radarhere** [comment]
Could you describe an example use case for this feature? The currently dictionary allows for the codes to be translated into something that can be understood by humans. What is your particular scenario where you want to go the other way?

**Comment by radarhere** [comment]
> A nicer solution would be to switch to an enum with the right values, but this is the quick win.

Could you expand on this?

**Comment by melonmouse** [comment]
Suppose a user wants to set an exif tag, I presume they would want to set the data using a human readable tag name.

**Comment by melonmouse** [comment]
> > A nicer solution would be to switch to an enum with the right values, but this is the quick win.
> 
> Could you expand on this?

Using a dict to store these values feels hacky in the first place.

```python
from enum import IntEnum


class ExifTags(IntEnum):
    InteropIndex = 0x0001
    ProcessingSoftware = 0x000B
    [...]
```

Which gives us just the right access:
```python
$ExifTags(0x0001).name
InteropIndex

$ExifTags.InteropIndex.value
1
```

**Comment by melonmouse** [comment]
To elaborate a bit more.

Currently:
```python
exif_data = Image.Exif()
exif_data[0x123456] = '2020:01:01 00:00:00'
image.save(filename, exif=exif_data.tobytes())
```

After this pullrequest:
```python
exif_data = Image.Exif()
exif_data[TAG_CODES['DateTimeOriginal']] = '2020:01:01 00:00:00'
image.save(filename, exif=exif_data.tobytes())
```

After bigger (breaking?) enum change:
```python
exif_data = Image.Exif()
exif_data[Exif.DateTimeOriginal] = '2020:01:01 00:00:00'
image.save(filename, exif=exif_data.tobytes())
```

(I did not test the above code, maybe I'm understanding how exif data is handled by PIL incorrectly)

**Comment by radarhere** [comment]
> Here your IDE will autocomplete the tagname. May need to update the code of ExifTags a bit to do this right. Could provide a TAGS dict generated from the Enum for backward compatibility.

I'm in favour of this, but when we previously had this conversation in #5875, I implemented it, [it was suggested that the original attributes be deprecated](https://github.com/python-pillow/Pillow/pull/5954#issuecomment-1009970094), and now there's #6537 where a user doesn't like that we changed the API. So I might hold off on this for a moment.

**Comment by melonmouse** [comment]
Whenever adding anything to an API, it is important to think about a potential future cost of deprecating it again.

Deprecation of the old approach would be costly indeed. It is always a balance between quality of a library and the amount of maintenance for existing users. It is never possible to please everyone, but we can try our best. For a change like the one suggested in this thread, backward compatibility should win. Adding the Enum in a backward compatible way is possible however. I'm sorry about not noticing the backward compatibility break in #5954.

In any case, thank you for your hard work on PIL!

**Comment by radarhere** [comment]
I've created PR #6630 to use enums as a possible alternative to this.

**Comment by radarhere** [comment]
#6630 has been merged.

**Pull request closed without merging**